nifi-api/src/main/java/org/apache/nifi/registry/MultiMap.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.registry;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class MultiMap<K,V> implements Map<K,V> {
+
+    private final List<Map<K,V>> maps;
+
+    MultiMap() {
+        this.maps = new ArrayList<>();
+    }
+
+    @Override
+    public int size() {
+        int size = 0;
+        for (final Map<K,V> map : maps) {
+            size += map.size();
+        }
+        return size;
```

Comment:
I'm not sure that this is accurate. If two of the maps were to have the same key, the size would be wrong here. Probably will need to create a Set<String> to store all keys in and then return the size of that?